core/src/main/java/org/apache/calcite/runtime/SqlFunctions.java

```diff
@@ -2747,6 +2747,14 @@ public static Object structAccess(Object structObject, int index, String fieldNa
       list = Arrays.asList(flatElements);
     }
 
+    @Override public boolean moveNext() {
```

Comment:
Well spotted!
I have not tested it, but I think we should probably override the reset method too, with something like:
```
    @Override public void reset() {
      super.reset();
      if (withOrdinality) {
        ordinality = 0;
      }
    }
```

Reply:
@rubenada Comments addressed. Thanks, Rubenada!